src/main/java/io/vertx/core/eventbus/Interceptor.java

```diff
@@ -0,0 +1,20 @@
+package io.vertx.core.eventbus;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Handler;
+
+/**
+ * An eventbus interceptor that will be called whenever a message is sent from, and delivered by,
+ * Vert.x
+ */
+@VertxGen
+public interface Interceptor extends Handler<SendContext> {
+  /**
+   * A message delivery attempt is about to occur.
+   *
+   * @param context the message delivery context to handle
+   */
+  default void handleDelivery(DeliveryContext context) {
```

Comment:
now I rethink about it, we should rather have 4 methods:

- sendHandler(Handler<DeliveryContext>)
- Handler<DeliveryContext> sendHandler() (with @ genignore)
- receiveHandler(Handler<DeliveryContext>)
- Handler<DeliveryContext> receiveHandler() (with @ genignore)

This way the interface is really @ genignore enabled